src/keywords/rangeVector.cpp

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// Copyright (c) 2023 Team Dissolve and contributors
+
+#include "keywords/rangeVector.h"
+#include "base/lineParser.h"
+#include "classes/coreData.h"
+
+RangeVectorKeyword::RangeVectorKeyword(std::vector<Range> &data) : KeywordBase(typeid(this)), data_(data) {}
+
+/*
+ * Data
+ */
+
+// Return the data vector
+std::vector<Range> &RangeVectorKeyword::data() { return data_; }
+const std::vector<Range> &RangeVectorKeyword::data() const { return data_; }
+
+/*
+ * Arguments
+ */
+
+// Deserialise from supplied LineParser, starting at given argument offset
+bool RangeVectorKeyword::deserialise(LineParser &parser, int startArg, const CoreData &coreData)
+{
+    // Loop over arguments
+    for (auto n = startArg; n < parser.nArgs(); n += 2)
+    {
+
+        // Check to see if value pair are numbers
+        if (DissolveSys::isNumber(parser.argsv(n)))
+        {
+            if (DissolveSys::isNumber(parser.argsv(n + 1)))
+            {
+                // If both are numbers, initialise new Range
+                data_.emplace_back(parser.argd(n + 1), parser.argd(n));
+            }
+            else
+            {
+                return Messenger::error("Value '{}' provided for keyword '{}' doesn't appear to be a number.\n",
+                                        parser.argsv(n + 1), name());
+            }
+        }
+        else
+        {
+            return Messenger::error("Value '{}' provided for keyword '{}' doesn't appear to be a number.\n", parser.argsv(n),
+                                    name());
+        }
```

Comment:
This nested `if` kinda sticks out a bit for me - I appreciate that you can explicitly report the offending value string this way, but perhaps this looks nicer:
```suggestion
        if (DissolveSys::isNumber(parser.argsv(n)) && DissolveSys::isNumber(parser.argsv(n + 1)))
        {
            data_.emplace_back(parser.argd(n + 1), parser.argd(n));
        }
        else
        {
            return Messenger::error("Range ({} to {}) given to keyword '{}' doesn't appear to be numerical.\n",
                                    parser.argsv(n), parser.argsv(n + 1), name());
        }
```

Reply:
What about:

```suggestion
        if (DissolveSys::isNumber(parser.argsv(n)) && DissolveSys::isNumber(parser.argsv(n + 1)))
        {
            data_.emplace_back(parser.argd(n + 1), parser.argd(n));
        }
        else
        {
            return Messenger::error("Value '{}' given to keyword '{}' doesn't appear to be numerical.\n",
                                    DissolveSys::isNumber(parser.argsv(n)) ? parser.argsv(n + 1) : parser.argsv(n), name());
        }
```

Reply:
Yeah, OK, let's go with that!  I also just noticed that your range is initialised with the two arguments in the "wrong" order (n+1, n)....